samza-core/src/main/java/org/apache/samza/util/EmbeddedTaggedRateLimiter.java

```diff
@@ -115,6 +115,8 @@ public void init(Context context) {
             int numTasks = jobModel.getContainers().values().stream()
                 .mapToInt(cm -> cm.getTasks().size())
                 .sum();
+            Preconditions.checkArgument(e.getValue() >= numTasks,
```

Comment:
Shall we just warn the user instead of throw exception for this case?
Also change line 120 to be:
double effectiveRate = (double) e.getValue() / numTasks;

Reply:
Make sense. Turning the calculation into double is the real fix.